applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -582,6 +582,81 @@ static int cloud_search_cmd(cJSON *root_obj)
 	return 0;
 }
 
+static int cloud_search_config(cJSON * const root_obj)
+{
+	int ret;
+	cJSON *state_obj = NULL;
+	cJSON *config_obj = NULL;
+	cJSON *channel_obj = NULL;
+	struct cmd *chan = NULL;
+	struct cmd const *const group = &group_cfg_set;
+	struct cloud_command found_config_item = {
+		.group = CLOUD_CMD_GROUP_CFG_SET
+	};
+
+	if (root_obj == NULL) {
+		return -EINVAL;
+	}
+
+	/* A delta update will have state */
+	state_obj = cJSON_GetObjectItem(root_obj, "state");
```

Comment:
This will be freed when the root object is deleted, right?

Reply:
correct, `state_obj` remains part of `root_obj` here, so `state_obj` does not need to be deleted.  `config_obj` gets detached, so it does get deleted.

Reply:
Thanks, confusing times with these cJSON functions :)